### DISCUSSION TITLE
RPI-0 as USB Gadget

### DISCUSSION THREAD

**Discussion by jwessel** [onlaj/Piano-LED-Visualizer#328] [Ideas]
This is the typical setup for getting wire connected MIDI
![image](https://user-images.githubusercontent.com/1044558/150617193-ca31e429-3a1f-486f-bf14-a0785e7e15c3.png)

I think we could add a fourth option to the documentation:
![image](https://user-images.githubusercontent.com/1044558/150617241-79dbf898-2058-4f07-8de0-545c3a0c52b3.png)

I am using a usb gadget with a modified USB cable and it is working quiet well.  No more dropped light events from Synthesia.

**Comment by onlaj** [comment]
You mean using the RPi in OTG mode? https://gist.github.com/gbaman/50b6cca61dd1c3f88f41

It was my initial plan when building the prototype 3 years ago. I can't remember why, but at some point I abandoned that idea, Something wasn't working right, maybe it was lack of modified cable.
I'm sure one of the reasons was that it will only work with the Zero.

**Reply by jwessel** [reply]
OTG works on any RPi Zero so long as you have the right cabling, the and the right drivers loaded.  The same is true for the RPi4 and the USB-C port, but then you have to power it different way, by a breakout cable,  

On the RPi Zero, I have it setup that I can use either the OTG USB Hub, or the USB micro A to USB A cable with the power connection cut.  I don't even have to power off the board. I can change the cables and the OTG sense line does the rest.   The power connection is cut in the custom cable to prevent any possibility of the VBUS connection providing voltage back to the connected computer

**Comment by psscin01** [comment]
This sounds like a solution to a problem that I have been having. Could you expand on this a bit more? I would like to circumvent using the Sevilla SoftMidi if possible.
